SkyEditor.IO/Binary/IReadOnlyBinaryDataAccessor.cs

```diff
@@ -80,162 +72,130 @@ public interface IReadOnlyBinaryDataAccessor
         /// <param name="length">Length of data to read</param>
         Task<byte[]> ReadArrayAsync(long index, int length);
 
-#if ENABLE_SPAN_AND_MEMORY
         /// <summary>
         /// Reads a subset of the available data
         /// </summary>
         /// <param name="index">Index of the desired data</param>
         /// <param name="length">Length of data to read</param>
         Task<ReadOnlyMemory<byte>> ReadMemoryAsync(long index, int length);
-#endif
-    }
 
-    public static class IReadOnlyBinaryDataAccessorExtensions
-    {
+        IReadOnlyBinaryDataAccessor Slice(long offset, long length)
+        {
+            return this switch
+            {
+                ReadOnlyBinaryDataAccessorReference reference => new ReadOnlyBinaryDataAccessorReference(reference, offset, length),
+                _ => new ReadOnlyBinaryDataAccessorReference(this, offset, length)
+            };
+        }
+
 #region Integer Reads
 
         /// <summary>
         /// Reads a signed 16 bit little endian integer
         /// </summary>
         /// <param name="offset">Offset of the integer to read.</param>
         /// <returns>The integer from the given location</returns>
-        public static Int16 ReadInt16(this IReadOnlyBinaryDataAccessor accessor, long offset)
-        {
-#if ENABLE_SPAN_AND_MEMORY
-            return BitConverter.ToInt16(accessor.ReadSpan(offset, 2));
-#else
-            return BitConverter.ToInt16(accessor.ReadArray(offset, 2), 0);
-#endif
-        }
+        short ReadInt16(long offset) => BinaryPrimitives.ReadInt16LittleEndian(ReadSpan(offset, sizeof(short)));
```

Comment:
When I first heard about default interface implementations, I was so happy, because that those extension methods could finally go away in favor of this.